### PR TITLE
Update to OpenTelemetry 0.18.0

### PR DESCRIPTION
**Pull request by akulnurislam**

### CODE REVIEW

File: newrelic/exporter_test.go [hunk 1]
```diff
@@ -165,7 +166,10 @@ func TestEndToEndTracer(t *testing.T) {
 		}
 		depth := numSpans - n
 		ctx, span := tracer.Start(ctx, fmt.Sprintf("Span %d", depth))
-		span.SetAttributes(label.Int("depth", depth))
+		span.SetAttributes(
+			attribute.Int("depth", depth),
+			semconv.ServiceNameKey.String(serviceName),
```
Comment (RichVanderwal) [inline newrelic/exporter_test.go:171]: Hi @akulnurislam ! This one has us scratching our heads a bit. What has changed to require us to add this serviceName in order for the test to pass?

Comment (akulnurislam) [reply newrelic/exporter_test.go:171]: @RichVanderwal ya hahaha, I saw `NewExporter` doesn't have any setAttributes.

If I look `NewExportPipeline` there is set attribute serviceName for otel controller.

https://github.com/newrelic/opentelemetry-exporter-go/blob/e563425c51c9e256b329c37e72192baac0396aca/newrelic/exporter.go#L111-L113

Comment (akulnurislam) [reply newrelic/exporter_test.go]: @RichVanderwal I've changed set attribute into `NewTracerProvider`

File: newrelic/exporter_test.go [hunk 2]
```diff
@@ -152,8 +154,11 @@ func TestEndToEndTracer(t *testing.T) {
 		t.Fatalf("failed to instantiate exporter: %v", err)
 	}
 
+	r := resource.NewWithAttributes(semconv.ServiceNameKey.String(serviceName))
+
 	tracerProvider := sdktrace.NewTracerProvider(
 		sdktrace.WithBatcher(e, sdktrace.WithBatchTimeout(15), sdktrace.WithMaxExportBatchSize(10)),
+		sdktrace.WithResource(r),
```
Comment (prmsrswt) [inline newrelic/exporter_test.go:161]: I think it'd be nice to modify the example app like this as well (adding ServiceName resource here: https://github.com/newrelic/opentelemetry-exporter-go/blob/master/examples/simple/main.go#L57).

**Review by prmsrswt — DISMISSED** [review]
Just a small non-blocking suggestion, everything else LGTM!

File: newrelic/exporter_test.go (hunk 2 above)
Comment (akulnurislam) [reply newrelic/exporter_test.go]: @prmsrswt sure, thanks

Comment (akulnurislam) [reply newrelic/exporter_test.go]: I think this version should go up, after the new release
https://github.com/newrelic/opentelemetry-exporter-go/blob/844f0922993392b95d57217a1f962e039fe2c1fa/examples/simple/go.mod#L7

Comment (gunturaf) [reply newrelic/exporter_test.go]: a new `go.opentelemetry.io/otel` v0.19.0 has been released, should this PR be updated to the latest one?

Comment (RichVanderwal) [reply newrelic/exporter_test.go]: Indeed. Merging this PR soon as a stepping stone to 0.19.0.

**Review by RichVanderwal — APPROVED** [review]
Thanks again, @akulnurislam !

**Comment by nr-swilloughby** [comment]
@akulnurislam @gunturaf @RichVanderwal @onprem @jodeev We have just released v0.18.0 of our Go OpenTelemetry Exporter which supports OpenTelemetry for Go v0.19.0. Let us know if you have any questions or comments.
Thanks!

**Pull request merged**